ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1836,6 +1829,34 @@ AP_ServoRelayEvents *GCS_MAVLINK_Copter::get_servorelayevents() const
     return &copter.ServoRelayEvents;
 }
 
+AP_AdvancedFailsafe *GCS_MAVLINK_Copter::get_advanced_failsafe() const
+{
+#if ADVANCED_FAILSAFE == ENABLED
+    return &copter.g2.afs;
+#else
+    return nullptr;
+#endif
+}
+
+MAV_RESULT GCS_MAVLINK_Copter::handle_flight_termination(const mavlink_command_long_t &packet) {
+    MAV_RESULT result = MAV_RESULT_FAILED;
+
+#if ADVANCED_FAILSAFE == ENABLED
+    if (GCS_MAVLINK::handle_flight_termination(packet) != MAV_RESULT_ACCEPTED) {
```

Comment:
Although I haven't reviewed this patch in much detail, this particular section where we ask the AFS to handle the flight termination, and if it says it hasn't (by returning MAV_RESULT_FAILED or equivalent) the vehicle code handles it .. this seems reasonable to me.  I probably would have made the return a true/false with false representing failure but this is a minor point and perhaps it was done this way to fit in better with other vehicles.